src/answers-umd.js

```diff
@@ -259,50 +259,29 @@ class Answers {
       if (parsedConfig.templateBundle) {
         this.renderer.init(parsedConfig.templateBundle);
       }
-
-      this._handlePonyfillCssVariables(
-        parsedConfig.disableCssVariablesPonyfill,
-        this._invokeOnReady.bind(this));
-      return this;
+    } else {
+      // Templates are currently downloaded separately from the CORE and UI bundle.
+      // Future enhancement is to ship the components with templates in a separate bundle.
+      this.templates = new DefaultTemplatesLoader(templates => {
+        this.renderer.init(templates);
+      });
     }
 
-    // Templates are currently downloaded separately from the CORE and UI bundle.
-    // Future enhancement is to ship the components with templates in a separate bundle.
-    this.templates = new DefaultTemplatesLoader(templates => {
-      this.renderer.init(templates);
-      this._handlePonyfillCssVariables(
-        parsedConfig.disableCssVariablesPonyfill,
-        this._invokeOnReady.bind(this));
+    const onReadyPromise = new Promise((resolve, reject) => {
+      this._handlePonyfillCssVariables(parsedConfig.disableCssVariablesPonyfill)
+        .then(this._masterSwitchApi.isDisabled())
+        .then((isDisabled) => {
+          if (!isDisabled) {
```

Comment:
do we need to reject if isDisabled is true?

Reply:
I didn't call reject because I was trying to match the original logic flow as closely as possible, however I think you're right. It should be called.